applications/nrf_desktop/src/modules/hid_state.c

```diff
@@ -1036,6 +1086,12 @@ static void connect(const void *subscriber_id, uint8_t report_id)
 
 		report_send(rd, false, true);
 	}
+
+	if (IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT) &&
```

Comment:
maybe we could find this `IS_ENABLED(CONFIG_DESKTOP_HID_REPORT_KEYBOARD_SUPPORT` behind the `get_keyboard_subscriber`?

Reply:
or maybe not if this `get_keyboard_subscriber` will become more generic...

Reply:
`get_keyboard_subscriber` was renamed to more generic `get_linked_subscriber(REPORT_ID_KEYBOARD_KEYS)` (as requested in previous comments)

Reply:
In and out report related to HID keyboard are tightly coupled. That makes creating more generic approach a bit problematic.